floodlight/models/kinematics.py

```diff
@@ -0,0 +1,102 @@
+import warnings
+from dataclasses import dataclass
+
+import numpy as np
+
+from floodlight.utils.types import Numeric
+from floodlight import Pitch, XY
+from floodlight.core.property import PlayerProperty
+
+
+@dataclass
+class ModelDistance:
+    """Class for calculating distances of players on the pitch.
+
+    Parameters
+    __________
+    pitch: Pitch
+        Pitch object of the data. Should at least include information about the data's
+        unit. If pitch.unit is 'percent', length and width of the pitch are also needed.
+
+    """
+
+    pitch: Pitch = None
+    length: Numeric = None
+    width: Numeric = None
+    unit: str = None
+
+    def __init__(self, pitch):
+        self.unit = pitch.unit
+        if pitch.length is not None:
+            self.length = pitch.length
+        if pitch.width is not None:
+            self.width = pitch.width
+        if pitch.unit is not None:
+            self.unit = pitch.unit
```

Comment:
why not just ``self.length = pitch.length``? It does not really matter for your handling later on of self.length is None or does not exist at all

Reply:
I wasn't sure how liberal the pitch object is with its attributes. So the idea was that as long as I know the unit of the pitch (given its m / cm), I can calculate distances. I don't need info about the pitch boundaries to do that. This might become handy when dealing with data from GPS systems where you don't have a fixed coordinate system to begin with.  But if those attributes are a necessity for pitch objects anyways than this doesn't apply.

Reply:
True, but implementation-wise I do not see what you gain from omitting the ``self.length`` attribute completely, instead of setting it to ``None`` -> you'll have to check if it's a numeral anyways when starting to compute distances. I'd prefer to keep it simple here, and upon computation set a necessary condition: what are the minimal information you need to do the trick? Query those, and if they are not met, raise an Error. This way you can avoid complicated "what if" scenarios during initialization. See also: https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/